src/main/kotlin/org/rust/ide/actions/RsAbstractMover.kt

```diff
@@ -0,0 +1,90 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.ide.actions
+
+import com.intellij.codeInsight.editorActions.moveUpDown.LineRange
+import com.intellij.codeInsight.editorActions.moveUpDown.StatementUpDownMover
+import com.intellij.openapi.editor.Editor
+import com.intellij.psi.PsiElement
+import com.intellij.psi.PsiFile
+import org.rust.ide.utils.findElementAtIgnoreWhitespaceAfter
+import org.rust.ide.utils.findElementAtIgnoreWhitespaceBefore
+import org.rust.lang.core.psi.*
+import org.rust.lang.core.psi.ext.RsMod
+import org.rust.lang.core.psi.ext.parentOfType
+
+val PsiElement.line: Int? get() = containingFile.viewProvider.document?.getLineNumber(textRange.startOffset)
+
+abstract class RsAbstractMover : StatementUpDownMover() {
+
+    abstract val listOfContainers: List<Class<out PsiElement>>
```

Comment:
Let's name it just `containers`. Usually it is not necessary to repeat the name of the type in the name of variable